### DISCUSSION TITLE
"Cannot find module" when building application with dependencies in packages

### DISCUSSION THREAD

**Discussion by chav-aniket** [vercel/turborepo#1853] [Help] [answered]
Hey I'm rather new to building backend typescript applications and am using turborepo for the first time. I'm trying to use SWC and have a package which is a dependency of my app, however when I try to build, while it successfully builds it doesn't run. Instead it complains "Cannot find module" for my package. My package is a production dependency, I am building it as well. 

The command I'm using to build my application with swc is `swc src -d dist`. Is there something specific I should be doing to bundly my packages which is different to what happens with other dependencies? I tried posting a question under the swc repo as well but got no response so thought to try my luck here :')

```
# folder structure
apps
└── microservice
    ├── dist
    └── src
packages
├── eslint-config-custom
├── utils
     ├── sub-1
     └── sub-2
└── tsconfig
```
I build my utils with swc -d dist sub-1 sub-2
```json
{
  "$schema": "https://json.schemastore.org/swcrc",
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false,
      "dynamicImport": true,
      "decorators": true,
      "importAssertions": true
    },
    "transform": {
      "decoratorMetadata": true
    },
    "target": "es2015",
    "loose": true,
    "externalHelpers": true
  },
  "module": {
    "type": "commonjs",
    // These are defaults.
    "strict": false,
    "strictMode": true,
    "lazy": false,
    "noInterop": false
  },
  "minify": false
}
```

**Comment by anthonyshew** [comment, marked as answer]
Check out the examples folder for some guidance on packaging!

https://github.com/vercel/turbo/tree/main/examples
